packages/account/src/account.ts

```diff
@@ -201,6 +201,19 @@ export class Account extends AbstractAccount {
   async fund<T extends TransactionRequest>(request: T, params: EstimatedTxParams): Promise<T> {
     const { addedSignatures, estimatedPredicates, requiredQuantities, updateMaxFee } = params;
 
+    const {
+      consensusParameters: {
+        txParameters: { maxInputs },
+      },
+    } = this.provider.getChain();
+
+    if (bn(request.inputs.length) > maxInputs) {
```

Comment:
@mvares Please move this validation to the end of the `while` loop that exists within the `fund` method.

Reply:
https://github.com/FuelLabs/fuels-ts/pull/2822/commits/786501a9a2b585686cded75f179398207df2c20b